chat-android/src/test/java/com/ably/chat/FlowEmitterTest.kt

```diff
@@ -0,0 +1,34 @@
+package com.ably.chat
+
+import kotlinx.coroutines.test.runTest
+import org.junit.Assert
+import org.junit.Test
+
+class FlowEmitterTest {
```

Comment:
This simply assertion test doesn't work. I think `Flows` are meant for specific requirement for broadcasting events and doesn't fit right into ably emitter pattern. Also, `Kotlin` doesn't provide flexible API around collectors, getting active collectors, cancelling them etc. Dangling collectors can also cause memory leaks. 
If we really want to implement (coroutine/scope based) emitters in kotlin, It's better to do it using ConcurrentList instead

Reply:
https://github.com/ably-labs/ably-chat-kotlin/pull/35/commits/2f12adb9a2e67f67e3f11e8d6e143f78d4c4aebb have fixed the test, but had to make `on` method blocking as well as add extra workaround to make flows work as expected